### PR TITLE
Language server pointer type support + add `DLLImport` test

### PR DESCRIPTION
**Pull request by csyonghe**
Also with Natvis improvements for AST.

Make executable tests able to load and use `slang-rt`, which is needed to test `DLLImport` feature.

### CODE REVIEW

File: source/core/slang-platform.cpp [hunk 1]
```diff
@@ -170,10 +177,11 @@ SLANG_COMPILE_TIME_ASSERT(E_OUTOFMEMORY == SLANG_E_OUT_OF_MEMORY);
 /* static */SlangResult SharedLibrary::loadWithPlatformPath(char const* platformFileName, Handle& handleOut)
 {
     handleOut = nullptr;
-
-	void* h = dlopen(platformFileName, RTLD_NOW | RTLD_GLOBAL);
-	if(!h)
-	{
+    if (strlen(platformFileName) == 0)
```
Comment (jsmall-nvidia) [inline source/core/slang-platform.cpp:180]: What does passing null or "" now mean?

File: source/slang/slang.natvis [hunk 2]
```diff
@@ -15,12 +15,15 @@
     <DisplayString Condition="decl != 0">{(*(*(Slang::DeclRefBase*)this).decl).nameAndLoc}</DisplayString>
     <Expand>
       <ExpandedItem>decl ? ($T1*)(decl) : ($T1*)0</ExpandedItem>
-      <Item Name="[Substitutions]:">"========================="</Item>
-      <LinkedListItems>
-        <HeadPointer>substitutions.substitutions.pointer</HeadPointer>
-        <NextPointer>outer.pointer</NextPointer>
-        <ValueNode>this</ValueNode>
-      </LinkedListItems>
+      <Synthetic Name="[Substitutions]">
```
Comment (jsmall-nvidia) [inline source/slang/slang.natvis:18]: Nice to have better natvis.

Guess I worry about amount of 'stuff' needed to be kept in sync. This is to work around the AST derived types not having a virtual function, such that vs knows the type?

**Review by jsmall-nvidia — APPROVED** [review]

File: source/core/slang-platform.cpp (hunk 1 above)
Comment (csyonghe) [reply source/core/slang-platform.cpp:180]: It means use RTLD_DEFAULT or load the function from the current process/module.

File: source/slang/slang.natvis (hunk 2 above)
Comment (csyonghe) [reply source/slang/slang.natvis:18]: Right. It will also save one expansion click since it will just directly display the derived type.

**Pull request merged**